### PR TITLE
migrate multi-tenancy perf-test to master branch

### PR DESCRIPTION
**Pull request by chenqianfzh**
it is verified working in both scale-out and scale-up kubemark clusters

### CODE REVIEW

File: perf-tests/clusterloader2/pkg/measurement/common/bundle/test_metrics.go [hunk 1]
```diff
@@ -41,8 +43,12 @@ func createTestMetricsMeasurement() measurement.Measurement {
 	if metrics.etcdMetrics, err = measurement.CreateMeasurement("EtcdMetrics"); err != nil {
 		klog.Errorf("%v: etcdMetrics creation error: %v", metrics, err)
 	}
-	if metrics.schedulingMetrics, err = measurement.CreateMeasurement("SchedulingMetrics"); err != nil {
-		klog.Errorf("%v: schedulingMetrics creation error: %v", metrics, err)
+	// We temporarily disable the metric collection on scheduler in scale-out scenarios,
```
Comment (pdgetrf) [inline perf-tests/clusterloader2/pkg/measurement/common/bundle/test_metrics.go:46]: what's the reason to disable?

**Review by pdgetrf — COMMENTED** [review]
Is this ready for final review or still WIP?

File: perf-tests/clusterloader2/pkg/measurement/common/bundle/test_metrics.go (hunk 1 above)
Comment (yb01) [reply perf-tests/clusterloader2/pkg/measurement/common/bundle/test_metrics.go:46]: I guess this is pending design on how to gather and summarize scheduler metric with multiple TPs ( tenants on each TP ). there should be at least issues to track this task --- it is a major change in scale out design to perf test metric that we need to support.

**Review by yb01 — APPROVED** [review]
/lgtm
except the comments on supporting "aggregated" scheduler metrics with multiple TPs.

File: perf-tests/clusterloader2/pkg/measurement/common/bundle/test_metrics.go (hunk 1 above)
Comment (chenqianfzh) [reply perf-tests/clusterloader2/pkg/measurement/common/bundle/test_metrics.go:46]: In scale-out clusters, The scheduler component is in a different partition from the resource partition. So the perf-test tool complains about failures  to get the scheduler pod. 

Changes are needed to enable scheduler metrics collection in scale-out perf tests. 

I opened https://github.com/CentaurusInfra/arktos/issues/930 to track this work.

**Comment by chenqianfzh** [comment]
> Is this ready for final review or still WIP?

It is ready to merge.

**Comment by chenqianfzh** [comment]
> /lgtm
> except the comments on supporting "aggregated" scheduler metrics with multiple TPs.

I opened issue https://github.com/CentaurusInfra/arktos/issues to track.

File: perf-tests/clusterloader2/pkg/chaos/nodes.go [hunk 2]
```diff
@@ -111,7 +111,7 @@ func (k *NodeKiller) pickNodes() ([]v1.Node, error) {
 		return nil, err
 	}
 
-	prometheusPods, err := client.ListPodsWithOptions(k.client, monitoringNamespace, metav1.ListOptions{
+	prometheusPods, err := client.ListPodsWithOptions(k.client, util.GetTenant(), monitoringNamespace, metav1.ListOptions{
```
Comment (vinaykul) [inline perf-tests/clusterloader2/pkg/chaos/nodes.go:114]: Does this still work for the default case (scale-up) where no tenant has been specified?

File: perf-tests/clusterloader2/pkg/framework/client/objects.go [hunk 3]
```diff
@@ -147,10 +147,10 @@ func RetryFunction(f func() error, options ...*ApiCallOptions) wait.ConditionFun
 }
 
 // ListPodsWithOptions lists the pods using the provided options.
-func ListPodsWithOptions(c clientset.Interface, namespace string, listOpts metav1.ListOptions) ([]apiv1.Pod, error) {
+func ListPodsWithOptions(c clientset.Interface, tenant, namespace string, listOpts metav1.ListOptions) ([]apiv1.Pod, error) {
```
Comment (vinaykul) [inline perf-tests/clusterloader2/pkg/framework/client/objects.go:150]: Same question here and elsewhere applicable - we should probably test it for scale-up scenario - deploy a regular cluster and run perf test as we do in the scale-up case.

File: perf-tests/clusterloader2/pkg/chaos/nodes.go (hunk 2 above)
Comment (chenqianfzh) [reply perf-tests/clusterloader2/pkg/chaos/nodes.go:114]: yes.

If when no env var of SCALEOUT_TEST_TENANT is specified, the function will return "system" tenant. 

This is also verfied via a test run.

File: perf-tests/clusterloader2/pkg/framework/client/objects.go (hunk 3 above)
Comment (chenqianfzh) [reply perf-tests/clusterloader2/pkg/framework/client/objects.go:150]: This code change has been verified in scale-up scenario.

**Pull request merged**